src/System.Security.Cryptography.X509Certificates/tests/Cert.cs

```diff
@@ -0,0 +1,159 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Xunit;
+
+namespace System.Security.Cryptography.X509Certificates.Tests
+{
+    //
+    // Helper class centralizes all loading of PFX's. Loading PFX's is a problem because of the key on disk that it creates and gets left behind
+    // if the certificate isn't properly disposed. As a result:
+    //
+    //    - PFX loading needs to be synchronized. If two threads race to load the same PFX, they may accidentally end up pointing to the same
+    //      key container. This means the first one that diposes deletes the key out from the other.
+    //
+    //    - Properly disposing PFX's imported into a X509Certificate2Collection is a pain because X509Certificate2Collection doesn't implement IDisposable.
+    //      To make this easier, we wrap these in an ImportedCollection class that does implement IDisposable.
+    //
+    // Though PFX's are the ones that really need this, it's fine to use this for ordinary DER data as well. It's probably better to use
+    // a consistent pattern throughout.
+    //
+    internal static class Cert
+    {
+        //
+        // The New() methods have an overload for each X509Certificate2 constructor that takes data.
+        //
+        public static X509Certificate2 New(byte[] rawData)
+        {
+            lock (SyncObject)
```

Comment:
> And unless we’re currently parallelizing the same test across processes

We are (more than one test set executes on the same machine in parallel)

> This isn’t something we can fix on our side, though.

Actually, we just need to regenerate the PFX files to not have that attribute.


So, my preference is to keep the collection disposer part of the change, drop the synchronization, and I'd prefer to leave the single-cert stuff alone in the tests (rather than using the new utility methods), but won't argue that last point overly strenuously.